createwindow/createwindow.go

```diff
@@ -0,0 +1,289 @@
+	cmdErrorHandler func(err error) error) (ChromeWindow, error) {
+
+	programstate, err := command.ExecuteProgram(cmd, cmdErrorHandler)
+	if err != nil {
+		return ChromeWindow{}, err
+	}
+
+	ctx, err := establishChromeConnection(programstate, chromeConnTimeout)
+	if err != nil {
+		return ChromeWindow{}, err
+	}
+
+	return ChromeWindow{programstate, ctx}, nil
+}
+
+func SetupChrome(chromeWindow ChromeWindow, URL string) error {
+	if err := chromedp.Run(chromeWindow.Ctx, chromedp.Navigate(URL)); err != nil {
+		return err
+	}
+	// TODO: use cookies to login if user allows.
+	return nil
+}
+
+// Layout has the x , y coordinates of top left corner and width and height of window.
+type Layout struct {
+	x, y uint32
+	w, h uint16
+}
+
+func WindowsLayout(screenInfo *xproto.ScreenInfo, n int) (chromeLayouts []Layout, inputwindow Layout) {
+	heightScreen := 0.8 * float64(screenInfo.HeightInPixels)
+	widthScreen := screenInfo.WidthInPixels
+	inputwindow.h, inputwindow.w = uint16(0.2*float64(screenInfo.HeightInPixels)), uint16(widthScreen)
+	inputwindow.y = uint32(heightScreen)
+
+	rows := int(n/4) + 1
+	columns := int(math.Ceil(float64(n) / float64(rows)))
+
+	var temp Layout
+	temp.h = uint16(int(heightScreen) / rows)
+	temp.w = uint16(int(widthScreen) / columns)
+
+	for i, r := 0, 0; r < rows; r++ {
+		temp.y = uint32(uint16(r) * temp.h)
+		for c := 0; c < columns && i < n; c++ {
+			temp.x = uint32(uint16(c) * temp.w)
+			chromeLayouts = append(chromeLayouts, temp)
+			i++
+		}
+	}
+	return chromeLayouts, inputwindow
+}
+
+func (s *Session) Newconn(displayString string) (*xproto.ScreenInfo, error) {
+	var err error
+	s.X, err = xgb.NewConnDisplay(displayString)
+	if err != nil {
+		return nil, err
+	}
+
+	setup := xproto.Setup(s.X)
+	screenInfo := setup.DefaultScreen(s.X)
+
+	return screenInfo, nil
+}
+
+func establishChromeConnection(programState *command.ProgramState, timeout int) (context.Context, error) {
+	wsURL, err := command.WsURL(programState, timeout)
+	if err != nil {
+		return nil, fmt.Errorf("Could not connect to the chrome window. Encountered error %s", err.Error())
+	}
+
+	if wsURL == "" {
+		return nil, errors.New("must specify -devtools-ws-url")
+	}
+
+	allocatorContext, _ := chromedp.NewRemoteAllocator(context.Background(), wsURL)
+
+	ctx, _ := chromedp.NewContext(allocatorContext)
+
+	return ctx, nil
+}
+
+// ForceQuit closes all open windows.
+func (s *Session) ForceQuit() {
+	programs := s.getChromeList()
+
+	fmt.Println("starting force quit")
+
+	for _, q := range programs {
+		if q.ToClose() == true {
+			q.Quit()
+		}
+	}
+	// Input Window is gracefully closed, closing closed window is okay.
+	s.InputWin.Quit()
+
+}
+
+func (s *Session) CreateXephyrWindow(layout Layout, display int, cmdErrorHandler func(err error) error) error {
+	displayString := fmt.Sprintf(":%d", display)
+	xephyr := command.ExternalCommand{
+		Path: "Xephyr",
+		Arg: []string{
+			displayString,
+			"-ac", "-screen",
```

Comment:
You combined the wrong two arguments?
-screen and its value should be together (since the value corresponds to the screen flag)
not -ac and -screen